### PR TITLE
nomad_1_3: init at 1.3.0

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

https://github.com/hashicorp/nomad/releases/tag/v1.3.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8634,7 +8634,7 @@ with pkgs;
 
   noip = callPackage ../tools/networking/noip { };
 
-  nomad = nomad_1_2;
+  nomad = nomad_1_3;
```
Comment (maxeaubrey) [inline pkgs/top-level/all-packages.nix:8637]:
```suggestion
  nomad = nomad_1_2;
```

**Review by amaxine — CHANGES_REQUESTED** [review]
We usually do not bump to the newest nomad release as soon as it is out, instead waiting until several point releases in to avoid any teething problems a new release might have. In the case of 1.3, I think it's super important we keep following that plan, considering:

> agent: The state database on both clients and servers will automatically migrate its underlying database on startup. Downgrading to a previous version of an agent after upgrading it to Nomad 1.3 is not supported.

This is the first time in a while Nomad doesn't have a downgrade path. Let's not force anyone onto the very first stable release immediately.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -8651,6 +8651,11 @@ with pkgs;
     inherit (linuxPackages) nvidia_x11;
     nvidiaGpuSupport = config.cudaSupport or false;
   };
+  nomad_1_3 = callPackage ../applications/networking/cluster/nomad/1.3.nix {
+    buildGoModule = buildGo118Module;
```
Comment (maxeaubrey) [inline pkgs/top-level/all-packages.nix:8655]: I was surprised by this change, double checking nomad 1.3 is still on go 1.17, see e.g. https://github.com/hashicorp/nomad/blob/v1.3.0/.go-version or CI pipelines. No mention in the changelog either.

**Comment by techknowlogick** [comment]
@maxeaubrey thanks for the feedback, I've updated per your suggestions :)

**Comment by maxeaubrey** [comment]
Thanks! And thanks for keeping nomad updated in nix, it's appreciated.

Can you please change the commit message to `nomad_1_3: init at 1.3.0`, to accurately reflect the changes?

**Comment by techknowlogick** [comment]
@maxeaubrey Thanks again :) I've just made the suggested changes.

**Pull request merged**